### PR TITLE
feat: GEO 1074 scanning file before backend upload

### PR DESCRIPTION
**Pull request by goemen**
# Description

Scanning file before backend upload

Fixes # ([issue](https://finrms.atlassian.net/browse/GEO-1074))

## Type of change

- [ ] New feature (non-breaking change which adds functionality)

## Checklist

- [ ] I have read the [CONTRIBUTING](CONTRIBUTING.md) doc
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have already been accepted and merged

## Further comments

---

Thanks for the PR!

Deployments, as required, will be available below:
- [Frontend](https://pay-transparency-pr-684-frontend.apps.silver.devops.gov.bc.ca)
- [Admin Frontend](https://pay-transparency-pr-684-admin-frontend.apps.silver.devops.gov.bc.ca)
- [Backend external API console](https://pay-transparency-pr-684-backend-external.apps.silver.devops.gov.bc.ca/api/V1/docs)

Please create PRs in draft mode.  Mark as ready to enable:
- [Analysis Workflow](https://github.com/bcgov/fin-pay-transparency/actions/workflows/analysis.yml)

After merge, new images are deployed in:
- [Merge Workflow](https://github.com/bcgov/fin-pay-transparency/actions/workflows/merge.yml)

### CODE REVIEW

File: clamav-service/src/app.ts [hunk 1]
```diff
@@ -112,28 +112,29 @@ const validateApiKey =
...
     if(!path){
       logger.error('File path is missing in the request body');
       res.status(400).send({
         message: 'File path is missing in the request body',
       });
     }
-    const filePath = fs.realpathSync(PATH.resolve(os.tmpdir(), path));
-    if (!filePath.startsWith(os.tmpdir())) {
+
+    if (!path.startsWith(os.tmpdir())) {
```
Comment (sukanya-rath) [inline clamav-service/src/app.ts:131]: Any reasons why you removed the real path sync code which fixed an important codeql warning ?

Comment (goemen) [reply clamav-service/src/app.ts:131]: `fs.realpathSync(PATH.resolve(os.tmpdir(), path))` results in the wrong location for the file

Comment (sukanya-rath) [reply clamav-service/src/app.ts:131]: can you put log statements - and do a dry run - I'd like to see the log statements

Comment (goemen) [reply clamav-service/src/app.ts:131]: Actually nevermind, I dont know how I got that from.

**Pull request merged**